src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/add-polling-source.component.spec.ts

```diff
@@ -0,0 +1,107 @@
+import { CUSTOM_ELEMENTS_SCHEMA, ChangeDetectionStrategy } from "@angular/core";
+import { Apollo } from "apollo-angular";
+import { ApolloTestingModule } from "apollo-angular/testing";
+import { FormBuilder, FormsModule, ReactiveFormsModule } from "@angular/forms";
+import { ComponentFixture, TestBed } from "@angular/core/testing";
+import { AddPollingSourceComponent } from "./add-polling-source.component";
+import { ActivatedRoute } from "@angular/router";
+import { NgbModal, NgbModalRef, NgbModule } from "@ng-bootstrap/ng-bootstrap";
+import { FinalYamlModalComponent } from "../final-yaml-modal/final-yaml-modal.component";
+import { AppDatasetCreateService } from "src/app/dataset-create/dataset-create.service";
+import { Observable } from "rxjs";
+import { SetPollingSourceSection } from "src/app/shared/shared.types";
+
+describe("AddPollingSourceComponent", () => {
+    let component: AddPollingSourceComponent;
+    let fixture: ComponentFixture<AddPollingSourceComponent>;
+    let modalService: NgbModal;
+    let modalRef: NgbModalRef;
+    let createDatasetService: AppDatasetCreateService;
+
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            declarations: [AddPollingSourceComponent],
+            imports: [
+                ApolloTestingModule,
+                ReactiveFormsModule,
+                FormsModule,
+                NgbModule,
+            ],
+            schemas: [CUSTOM_ELEMENTS_SCHEMA],
```

Comment:
Remove schema and list all required components/modules in the dependencies.
Otherwise, we are not really testing how real components interact with each other.

Reply:
Ok, removed.